### PR TITLE
feat(provider/dcos): Add DC/OS server group wizard

### PR DESCRIPTION
**Pull request by willgorman**
Adds the wizard for  configuring DC/OS server groups.  This ended up being fairly large so we can split this up further if preferred.

### CODE REVIEW

File: app/scripts/modules/dcos/serverGroup/configure/wizard/templateSelection.controller.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
+'use strict';
```
Comment (anotherchrisberry) [inline app/scripts/modules/dcos/serverGroup/configure/wizard/templateSelection.controller.js]: We consolidated all the provider-specific template selectors a few weeks ago to use the `deployInitializer` component in core - can you give that a shot and see if it works?

**Comment by anotherchrisberry** [comment]
One question around the template selection. I didn't dig into the code beyond that, but assume it's good.

**Comment by willgorman** [comment]
The deployInitializer component did work for us, thanks.

**Comment by anotherchrisberry** [comment]
Awesome, glad to hear it

**Pull request merged**